rpcserver.go

```diff
@@ -1357,6 +1362,114 @@ func (r *rpcServer) ListTransfers(ctx context.Context,
 	return resp, nil
 }
 
+// SetPendingTransferProofCourierAddr sets the transfer output proof courier
+// address for a pending (undelivered proof) transfer. The anchoring transaction
+// of the target transfer must have already been confirmed on-chain.
+func (r *rpcServer) SetPendingTransferProofCourierAddr(ctx context.Context,
+	req *updateCourierAddrReq) (*updateCourierAddrResp, error) {
+
+	// Unmarshal the anchor tx hash.
+	if len(req.AnchorTxid) == 0 {
+		return nil, fmt.Errorf("anchor txid must be set")
+	}
+
+	anchorTxHash, err := chainhash.NewHashFromStr(req.AnchorTxid)
+	if err != nil {
+		return nil, fmt.Errorf("invalid anchor tx hash: %w", err)
+	}
+
+	// Validate the requested new proof courier address.
+	_, err = proof.ParseCourierAddress(req.NewProofCourierAddr)
+	if err != nil {
+		return nil, fmt.Errorf("invalid proof courier address: %w", err)
+	}
+
+	// Unmarshal the transfer output script key public key.
+	outputScriptKey, err := parseUserKey(req.TransferOutputScriptPubKey)
+	if err != nil {
+		return nil, fmt.Errorf("invalid script pub key: %w", err)
+	}
+
+	// Query for pending transfers only.
+	pendingParcels, err := r.cfg.AssetStore.QueryParcels(
+		ctx, anchorTxHash, true,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to query parcels: %w", err)
+	}
+
+	// Identify the target parcel and modify the proof courier address.
+	var targetParcel *tapfreighter.OutboundParcel
+	for idxParcel := range pendingParcels {
+		pendingParcel := pendingParcels[idxParcel]
+
+		// The anchoring transaction of target parcel must have already
+		// been confirmed on-chain.
+		if pendingParcel.AnchorTxBlockHash.IsNone() {
+			continue
```

Comment:
Does this matter? Maybe we want to update before we reached a confirmation?
I assume it has something to do with how we insert a new parcel, see comment below.